api/org/commcare/api/persistence/SqlHelper.java

```diff
@@ -0,0 +1,224 @@
+package org.commcare.api.persistence;
+
+import org.commcare.modern.util.Pair;
+import org.commcare.modern.database.*;
+import org.javarosa.core.services.storage.Persistable;
+
+import java.io.ByteArrayInputStream;
+import java.sql.Blob;
+import java.sql.Connection;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.util.HashMap;
+import java.util.List;
+
+/**
+ * Set of Sql utility methods for clients running modern, non-Android Java (where we use cursors
+ * in the place of prepared statements)
+ *
+ * Created by wpride1 on 8/11/15.
+ */
+public class SqlHelper {
+    public static void dropTable(Connection c, String storageKey) {
+        String sqlStatement = "DROP TABLE IF EXISTS " + storageKey;
+        try {
+            PreparedStatement preparedStatement = c.prepareStatement(sqlStatement);
+            preparedStatement.execute();
+        } catch (SQLException e) {
+            System.out.println("didn't drop table");
+        }
+    }
+
+    public static ResultSet executeSql(Connection c, String sqlQuery) {
+        try {
+            PreparedStatement preparedStatement = c.prepareStatement(sqlQuery);
+            return preparedStatement.executeQuery();
+        } catch (SQLException e) {
+            e.printStackTrace();
+            return null;
+        }
+    }
+
+    public static byte[] getRecordForValue(Connection c, String storageKey, String id) {
+        String sqlStatement = DatabaseHelper.getRecordForValueSelectString(storageKey, id);
+        try {
+            PreparedStatement preparedStatement = c.prepareStatement(sqlStatement);
+            ResultSet rs = preparedStatement.executeQuery();
+            return rs.getBytes(1);
+        } catch (SQLException e) {
+            e.printStackTrace();
+            return null;
+        }
+    }
+
+    public static void createTable(Connection c, String storageKey, Persistable p) {
+        String sqlStatement = DatabaseHelper.getTableCreateString(storageKey, p);
+        PreparedStatement preparedStatement = null;
+        try {
+            preparedStatement = c.prepareStatement(sqlStatement);
+            preparedStatement.execute();
+        } catch (SQLException e) {
+            //fail silently, expected sometimes
+        } finally {
+            if (preparedStatement != null) {
+                try {
+                    preparedStatement.close();
+                } catch (SQLException e) {
+
+                }
```

Comment:
You will probably want to execute sql in a `try` and close the statement safely a lot. I recommend writing a helper function that takes in a statement, and safely executes and closes it, so that you don't have to repeat the 10 lines of try/catch/finally code.